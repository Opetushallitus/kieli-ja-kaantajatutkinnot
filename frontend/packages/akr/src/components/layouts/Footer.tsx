import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { Divider, Paper } from '@mui/material';
import {
  CustomButtonLink,
  ExtLink,
  H2,
  OPHLogoViewer,
  Svg,
  Text,
} from 'shared/components';
import { Direction, Variant } from 'shared/enums';
import { FooterWave } from 'shared/statics';

import {
  getCurrentLang,
  useAppTranslation,
  useCommonTranslation,
} from 'configs/i18n';
import { useAppSelector } from 'configs/redux';
import { AppRoutes, PublicUIViews } from 'enums/app';
import { useAuthentication } from 'hooks/useAuthentication';
import AKTLogo from 'public/assets/svg/akt_logo.svg';
import { publicUIViewSelector } from 'redux/selectors/publicUIView';

export const Footer = () => {
  const { t } = useAppTranslation({ keyPrefix: 'akr.component.footer' });
  const translateCommon = useCommonTranslation();
  const { isAuthenticated } = useAuthentication();
  const { currentView } = useAppSelector(publicUIViewSelector);
  const showFooter =
    !isAuthenticated && currentView != PublicUIViews.ContactRequest;

  return (
    <footer>
      {showFooter && (
        <>
          <Svg
            className="footer__wave"
            src={FooterWave}
            alt={t('accessibility.waveAriaLabel')}
          />
          <Paper className="footer" elevation={3}>
            <div className="footer__info-row">
              <div className="footer__container footer__container__links">
                <CustomButtonLink
                  to={AppRoutes.AccessibilityStatementPage}
                  variant={Variant.Text}
                >
                  {t('links.accessibility.text')}
                </CustomButtonLink>
                <CustomButtonLink
                  to={AppRoutes.PrivacyPolicyPage}
                  variant={Variant.Text}
                >
                  {t('links.privacy.text')}
                </CustomButtonLink>
                <ExtLink
                  text={t('links.akrHomepage.text')}
                  href={t('links.akrHomepage.link')}
                  endIcon={<OpenInNewIcon />}
                  aria-label={t('links.akrHomepage.ariaLabel')}
                />
                <div className="footer__container__links__contact">
                  <H2>{t('links.contact.title')}:</H2>
                  <ExtLink
                    className="footer__container__links__contact__email"
                    href={`mailto:${translateCommon('contactEmail')}`}
                    text={translateCommon('contactEmail')}
                  ></ExtLink>
                </div>
              </div>
              <div className="footer__container footer__container__contact-details">
                <H2>{t('address.name')}</H2>
                <br />
                <Text>{t('address.street')}</Text>
                <Text>{t('address.zipCity')}</Text>
                <br />
                <div className="columns gapped-xxs">
                  <Text className="inline-text">
                    {t('address.phone.title')}
                  </Text>
                  <ExtLink
                    className="inline-text"
                    text={t('address.phone.number')}
                    href={`tel:${t('address.phone.number')}`}
                    aria-label={t('accessibility.ophPhone')}
                  />
                </div>
              </div>
              <div className="footer__container">
                <Svg
                  className={'footer__container__logo__akr'}
                  src={AKTLogo}
                  alt={translateCommon('akrLogo')}
                />
              </div>
            </div>

            <div className="footer__logo-row">
              <Divider className="footer__logo-row__divider">
                <OPHLogoViewer
                  className="footer__container__logo__oph"
                  direction={Direction.Vertical}
                  alt={translateCommon('ophLogo')}
                  currentLang={getCurrentLang()}
                />
              </Divider>
            </div>
          </Paper>
        </>
      )}
    </footer>
  );
};
