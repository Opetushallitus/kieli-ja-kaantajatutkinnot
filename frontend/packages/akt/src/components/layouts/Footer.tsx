import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { Divider, Paper } from '@mui/material';

import { CustomButtonLink } from 'components/elements/CustomButtonLink';
import { ExtLink } from 'components/elements/ExtLink';
import { OPHLogoViewer } from 'components/elements/OPHLogoViewer';
import { Svg } from 'components/elements/Svg';
import { H3, Text } from 'components/elements/Text';
import { useAppTranslation, useCommonTranslation } from 'configs/i18n';
import { AppRoutes, Direction, Variant } from 'enums/app';
import { useAuthentication } from 'hooks/useAuthentication';
import AKTLogo from 'public/assets/svg/akt_logo.svg';
import FooterWave from 'public/assets/svg/footer_wave.svg';

export const Footer = () => {
  const { t } = useAppTranslation({ keyPrefix: 'akt.component.footer' });
  const translateCommon = useCommonTranslation();
  const [isClerkUI] = useAuthentication();

  return (
    <footer>
      {!isClerkUI && (
        <>
          <Svg
            className="footer__wave"
            src={FooterWave}
            alt={t('accessibility.waveAriaLabel')}
          />
          <Paper className="footer" elevation={3}>
            <div className="footer__info-row">
              <div className="footer__container footer__container--links">
                <CustomButtonLink
                  to={AppRoutes.AccessibilityStatementPage}
                  variant={Variant.Text}
                >
                  {t('links.accessibility.text')}
                </CustomButtonLink>
                <ExtLink
                  text={t('links.aktHomepage.text')}
                  href={t('links.aktHomepage.link')}
                  endIcon={<OpenInNewIcon />}
                  aria-label={t('links.aktHomepage.ariaLabel')}
                />
                <ExtLink
                  className="footer__container--links__contact-email"
                  href={`mailto:${translateCommon('contactEmail')}`}
                  text={translateCommon('contactEmail')}
                ></ExtLink>
              </div>
              <div className="footer__container footer__container--contact-details">
                <H3>{t('address.name')}</H3>
                <br />
                <Text>{t('address.street')}</Text>
                <Text>{t('address.zipCity')}</Text>
                <br />
                <div className="columns gapped-xxs">
                  <Text className="inline-text">
                    {t('address.phone.title')}
                  </Text>
                  <ExtLink
                    className="inline-text"
                    text={t('address.phone.number')}
                    href={`tel:${t('address.phone.number')}`}
                    aria-label={t('accessibility.ophPhone')}
                  />
                </div>
              </div>
              <div className="footer__container">
                <Svg
                  className={'footer__container__logo--akt'}
                  src={AKTLogo}
                  alt={translateCommon('aktLogo')}
                />
              </div>
            </div>

            <div className="footer__logo-row">
              <Divider className="footer__logo-row__divider">
                <OPHLogoViewer
                  className="footer__container__logo--oph"
                  direction={Direction.Vertical}
                />
              </Divider>
            </div>
          </Paper>
        </>
      )}
    </footer>
  );
};
