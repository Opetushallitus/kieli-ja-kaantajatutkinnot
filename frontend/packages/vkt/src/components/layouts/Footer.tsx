import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { Divider, Paper, Typography } from '@mui/material';
import { useLocation } from 'react-router-dom';
import {
  CustomButtonLink,
  ExtLink,
  OPHLogoViewer,
  Svg,
  Text,
} from 'shared/components';
import { Direction, Variant } from 'shared/enums';
import { FooterWave } from 'shared/statics';

import {
  getCurrentLang,
  useCommonTranslation,
  usePublicTranslation,
} from 'configs/i18n';
import { AppRoutes } from 'enums/app';
import { useAuthentication } from 'hooks/useAuthentication';

export const Footer = () => {
  const location = useLocation();

  const { t } = usePublicTranslation({ keyPrefix: 'vkt.component.footer' });
  const translateCommon = useCommonTranslation();

  const { isAuthenticated } = useAuthentication();
  const showFooter =
    !isAuthenticated &&
    ![AppRoutes.PublicEnrollment, AppRoutes.PublicAuth].includes(
      location.pathname as AppRoutes
    );

  return (
    <footer>
      {showFooter && (
        <>
          <Svg className="footer__wave" src={FooterWave} alt="" />
          <Paper className="footer" elevation={3}>
            <div className="footer__info-row">
              <div className="footer__container footer__container__links">
                <CustomButtonLink
                  to={AppRoutes.AccessibilityStatementPage}
                  variant={Variant.Text}
                >
                  {t('links.accessibility.text')}
                </CustomButtonLink>
                <CustomButtonLink
                  to={AppRoutes.PrivacyPolicyPage}
                  variant={Variant.Text}
                >
                  {t('links.privacy.text')}
                </CustomButtonLink>
                <ExtLink
                  text={t('links.vktHomepage.text')}
                  href={translateCommon('vktHomepage.link')}
                  endIcon={<OpenInNewIcon />}
                  aria-label={translateCommon('vktHomepage.ariaLabel')}
                />
                <div className="footer__container__links__contact">
                  <Typography component="h2" variant="h3">
                    {t('links.contact.title')}:
                  </Typography>
                  <ExtLink
                    className="footer__container__links__contact__email"
                    href={`mailto:${translateCommon('contactEmail')}`}
                    text={translateCommon('contactEmail')}
                  ></ExtLink>
                </div>
              </div>
              <div className="footer__container footer__container__contact-details">
                <Typography component="h2" variant="h3">
                  {t('address.name')}
                </Typography>
                <br />
                <Text>{t('address.street')}</Text>
                <Text>{t('address.zipCity')}</Text>
                <br />
                <div className="columns gapped-xxs">
                  <Text className="inline-text">
                    {t('address.phone.title')}
                  </Text>
                  <ExtLink
                    className="inline-text"
                    text={t('address.phone.number')}
                    href={`tel:${t('address.phone.number')}`}
                  />
                </div>
              </div>
            </div>

            <div className="footer__logo-row">
              <Divider className="footer__logo-row__divider">
                <OPHLogoViewer
                  className="footer__container__logo__oph"
                  direction={Direction.Vertical}
                  alt={translateCommon('ophLogoAlt')}
                  currentLang={getCurrentLang()}
                />
              </Divider>
            </div>
          </Paper>
        </>
      )}
    </footer>
  );
};
