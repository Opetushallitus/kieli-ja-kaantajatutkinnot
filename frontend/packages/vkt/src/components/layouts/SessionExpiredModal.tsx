import { CustomButton, CustomModal, Text } from 'shared/components';
import { Color, Variant } from 'shared/enums';

import { useCommonTranslation } from 'configs/i18n';
import { AppRoutes } from 'enums/app';

export const SessionExpiredModal = ({ isClerkUI }: { isClerkUI: boolean }) => {
  const translateCommon = useCommonTranslation();

  return (
    <CustomModal
      data-testid="public-enrollment__session-expired-modal"
      className="public-enrollment__session-expired-modal"
      open={true}
      aria-labelledby="session-expired-modal-title"
      aria-describedby="session-expired-modal-description"
      modalTitle={translateCommon('sessionExpired.title')}
      onCloseModal={() => {}}
    >
      <>
        <Text id="session-expired-modal-description">
          {translateCommon(
            isClerkUI
              ? 'sessionExpired.descriptionClerk'
              : 'sessionExpired.descriptionPublic',
          )}
        </Text>
        <div className="columns gapped flex-end">
          <CustomButton
            data-testid="public-enrollment__session-expired-modal-button"
            variant={Variant.Contained}
            color={Color.Secondary}
            href={AppRoutes.PublicHomePage}
          >
            {translateCommon('backToHomePage')}
          </CustomButton>
        </div>
      </>
    </CustomModal>
  );
};
