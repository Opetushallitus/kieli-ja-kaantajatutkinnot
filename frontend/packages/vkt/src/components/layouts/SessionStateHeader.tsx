import {
  LogoutOutlined as LogoutIcon,
  Person as PersonIcon,
} from '@mui/icons-material';
import Button from '@mui/material/Button';
import { FC } from 'react';
import { Text } from 'shared/components';
import { Color, Variant } from 'shared/enums';

import { useCommonTranslation } from 'configs/i18n';
import { useAppDispatch } from 'configs/redux';
import { APIEndpoints } from 'enums/api';
import { useInterval } from 'hooks/useInterval';
import { loadPublicUser } from 'redux/reducers/publicUser';

interface SessionStateHeaderProps {
  firstName: string;
  lastName: string;
}

export const SessionStateHeader: FC<SessionStateHeaderProps> = ({
  firstName,
  lastName,
}) => {
  const dispatch = useAppDispatch();
  const translateCommon = useCommonTranslation();

  const heartBeat = () => {
    if (!document.hidden) {
      dispatch(loadPublicUser());
    }
  };

  useInterval(heartBeat, 5000); // Every 5 seconds

  return (
    <div className="session-header columns gapped">
      <Text className="columns gapped-xxs session-header__user-details">
        <PersonIcon className="session-header__user-icon" />
        {firstName} {lastName}
      </Text>
      <a href={APIEndpoints.PublicAuthLogout}>
        <Button color={Color.Secondary} variant={Variant.Outlined}>
          <LogoutIcon />
          {translateCommon('header.sessionState.logOut')}
        </Button>
      </a>
    </div>
  );
};
