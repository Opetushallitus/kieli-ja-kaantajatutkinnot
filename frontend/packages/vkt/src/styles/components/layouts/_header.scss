.header {
  &__toolbar {
    // stylelint-disable-next-line selector-class-pattern
    &.MuiToolbar-root {
      @include not-phone {
        display: grid;
        grid-template-areas:
          'logo language-select'
          'logo navigation';
        grid-auto-columns: min-content auto;
      }
      flex-wrap: wrap;
    }

    div &__mobile-lang-select {
      min-height: fit-content;
      padding: 0;
    }
  }

  &__logo button,
  &__navigation button,
  &__language-select button {
    @include header-text;
  }

  &__logo {
    @include not-phone {
      grid-area: logo;
    }
    margin-right: 5rem;
    padding: 2rem 0;
    @include phone {
      margin-right: 0;
    }

    & > a {
      text-decoration: none;
    }

    &__logo {
      height: 5.5rem;
      @include phone {
        height: 5rem;
      }
    }
  }

  &__navigation {
    @include not-phone {
      grid-area: navigation;
    }
    align-self: flex-end;
    justify-self: start;
  }

  &__language-select {
    @include not-phone {
      grid-area: language-select;
    }
    @include phone {
      align-self: center;
    }
    align-self: center;
    color: $color-secondary-dark;
    display: flex;
    flex-grow: 1;
    gap: 2rem;
    justify-content: flex-end;

    & &__avatar {
      background-color: $color-secondary-dark;
      font-size: 1.1rem;
      height: 2rem;
      margin-right: 0.3rem;
      width: 2rem;
    }

    & > a {
      @include header-text;
    }
  }
}
