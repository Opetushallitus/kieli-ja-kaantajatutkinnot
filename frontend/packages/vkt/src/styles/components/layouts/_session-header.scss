.session-header {
  background: #f0f3f7;
  justify-content: flex-end;
  padding: 2rem 6rem 2rem 2rem;

  @include phone {
    align-items: flex-end;
    flex-direction: column;
    gap: 0;
    justify-content: center;
    padding: 1rem 2rem 1rem 1rem;
  }

  & &__user-icon {
    height: 2rem;
    width: 2rem;
  }
}
