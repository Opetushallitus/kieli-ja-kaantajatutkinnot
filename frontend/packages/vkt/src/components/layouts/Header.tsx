import { AppBar, Toolbar } from '@mui/material';
import { Link, matchPath, useLocation } from 'react-router-dom';
import {
  CookieBanner,
  LangSelector,
  NavigationLinks,
  OPHClerkLogo,
  OPHLogoViewer,
  SkipLink,
  Text,
} from 'shared/components';
import { APIResponseStatus, AppLanguage, Direction } from 'shared/enums';
import { useWindowProperties } from 'shared/hooks';

import { ClerkHeaderButtons } from 'components/layouts/clerkHeader/ClerkHeaderButtons';
import { ClerkNavTabs } from 'components/layouts/clerkHeader/ClerkNavTabs';
import { SessionExpiredModal } from 'components/layouts/SessionExpiredModal';
import { SessionStateHeader } from 'components/layouts/SessionStateHeader';
import {
  changeLang,
  getCurrentLang,
  getSupportedLangs,
  useCommonTranslation,
} from 'configs/i18n';
import { useAppDispatch, useAppSelector } from 'configs/redux';
import { AppRoutes, PublicNavigationLink } from 'enums/app';
import { useAuthentication } from 'hooks/useAuthentication';
import { useInterval } from 'hooks/useInterval';
import { loadClerkUser } from 'redux/reducers/clerkUser';
import { loadPublicUser } from 'redux/reducers/publicUser';
import { featureFlagsSelector } from 'redux/selectors/featureFlags';

const isPathActive = (currentPath: string, route: AppRoutes) =>
  !!matchPath({ path: route, end: false }, currentPath);

const PublicNavigationLinks = () => {
  const translateCommon = useCommonTranslation();
  const { pathname } = useLocation();
  const { goodAndSatisfactoryLevel } = useAppSelector(featureFlagsSelector);
  const excellentLevelLink = {
    active: isPathActive(pathname, AppRoutes.PublicExcellentLevelLanding),
    label: translateCommon(
      `header.publicNavigationLinks.${PublicNavigationLink.ExcellentLevel}`,
    ),
    href: AppRoutes.PublicExcellentLevelLanding,
  };

  const navigationLinks = goodAndSatisfactoryLevel
    ? [
        {
          active: isPathActive(pathname, AppRoutes.PublicHomePage),
          label: translateCommon(
            `header.publicNavigationLinks.${PublicNavigationLink.FrontPage}`,
          ),
          href: AppRoutes.PublicHomePage,
        },
        excellentLevelLink,
        {
          active: isPathActive(
            pathname,
            AppRoutes.PublicGoodAndSatisfactoryLevelLanding,
          ),
          label: translateCommon(
            `header.publicNavigationLinks.${PublicNavigationLink.GoodAndSatisfactoryLevel}`,
          ),
          href: AppRoutes.PublicGoodAndSatisfactoryLevelLanding,
        },
      ]
    : [excellentLevelLink];

  return (
    <NavigationLinks
      navigationAriaLabel={translateCommon(
        'header.accessibility.mainNavigation',
      )}
      links={navigationLinks}
    />
  );
};

export const Header = (): JSX.Element => {
  const dispatch = useAppDispatch();
  const translateCommon = useCommonTranslation();
  const [finnish, swedish] = getSupportedLangs();

  const langDict = new Map<string, AppLanguage>([
    [translateCommon('header.lang.fi'), finnish],
    [translateCommon('header.lang.sv'), swedish],
  ]);

  const { isAuthenticated, isClerkUI, clerkUser, publicUser } =
    useAuthentication();
  const goodAndSatisfactoryLevelSupported =
    useAppSelector(featureFlagsSelector).goodAndSatisfactoryLevel;
  const logoRedirectURL = isAuthenticated
    ? AppRoutes.ClerkHomePage
    : AppRoutes.PublicHomePage;
  const activeUrl = window.location.href;
  const isPublicUrl = !activeUrl.includes(AppRoutes.ClerkHomePage);
  const { isPhone } = useWindowProperties();

  const isClerkAuthenticationValid =
    clerkUser.status == APIResponseStatus.NotStarted ||
    clerkUser.status == APIResponseStatus.InProgress ||
    clerkUser.isAuthenticated;

  const heartBeat = () => {
    if (isAuthenticated || publicUser?.isAuthenticated) {
      if (!document.hidden) {
        dispatch(isClerkUI ? loadClerkUser() : loadPublicUser());
      }
    }
  };

  useInterval(heartBeat, 5000); // Every 5 seconds

  return (
    <>
      {isClerkUI && !isClerkAuthenticationValid && (
        <SessionExpiredModal isClerkUI />
      )}
      <SkipLink
        href="#main-content"
        text={translateCommon('header.accessibility.continueToMain')}
      />
      <AppBar className="header" position="static">
        {isPhone && (
          <Toolbar className="header__toolbar header__toolbar__mobile-lang-select">
            <LangSelector
              changeLang={changeLang}
              getCurrentLang={getCurrentLang}
              langDict={langDict}
              langSelectorAriaLabel={translateCommon(
                'header.accessibility.langSelectorAriaLabel',
              )}
            />
          </Toolbar>
        )}
        {publicUser?.isAuthenticated && (
          <SessionStateHeader
            firstName={publicUser.firstName}
            lastName={publicUser.lastName}
          />
        )}
        <Toolbar className="header__toolbar">
          <div className="header__left">
            <Link to={logoRedirectURL}>
              {isClerkUI ? (
                <OPHClerkLogo
                  mainLabel={translateCommon('appNameAbbreviation')}
                  subLabel={translateCommon('clerk')}
                  alt={translateCommon('ophLogoToFrontPageAlt')}
                />
              ) : (
                <OPHLogoViewer
                  className="header__left__logo"
                  direction={Direction.Horizontal}
                  alt={translateCommon('ophLogoToFrontPageAlt')}
                  currentLang={getCurrentLang()}
                  title={translateCommon('appNameAbbreviation')}
                />
              )}
            </Link>
          </div>
          <div className="header__center">
            {isAuthenticated && <ClerkNavTabs />}
            {isPublicUrl && goodAndSatisfactoryLevelSupported && (
              <PublicNavigationLinks />
            )}
          </div>
          <div className="header__right">
            {isAuthenticated && <ClerkHeaderButtons />}
            {!isPhone && (
              <LangSelector
                changeLang={changeLang}
                getCurrentLang={getCurrentLang}
                langDict={langDict}
                langSelectorAriaLabel={translateCommon(
                  'header.accessibility.langSelectorAriaLabel',
                )}
              />
            )}
          </div>
        </Toolbar>
      </AppBar>
      {!isClerkUI && (
        <CookieBanner
          title={translateCommon('cookieBanner.title')}
          buttonText={translateCommon('cookieBanner.buttonText')}
          cookieTag="cookie-consent-vkt"
          buttonAriaLabel={translateCommon('cookieBanner.buttonAriaLabel')}
          path="/vkt"
        >
          <Text data-testid="cookie-banner-description">
            {translateCommon('cookieBanner.description')}
          </Text>
        </CookieBanner>
      )}
    </>
  );
};
