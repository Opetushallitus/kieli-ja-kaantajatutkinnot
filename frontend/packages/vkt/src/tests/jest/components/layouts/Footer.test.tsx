import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { Footer } from 'components/layouts/Footer';

describe('Footer', () => {
  it('should render Footer correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Footer />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
