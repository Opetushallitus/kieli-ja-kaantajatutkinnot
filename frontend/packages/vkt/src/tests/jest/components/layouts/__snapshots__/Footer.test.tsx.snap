// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render Footer correctly 1`] = `
<footer>
  <img
    alt=""
    className="footer__wave"
    src="test"
  />
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 footer css-3su884-MuiPaper-root"
  >
    <div
      className="footer__info-row"
    >
      <div
        className="footer__container footer__container__links"
      >
        <a
          href="/vkt/saavutettavuusseloste"
          onClick={[Function]}
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            links.accessibility.text
          </p>
        </a>
        <a
          aria-label="vktPrivacyPolicy.ariaLabel"
          className="columns gapped-xxs"
          href="vktPrivacyPolicy.link"
          rel="noreferrer"
          target="_blank"
        >
          links.privacy.text
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="OpenInNewIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
            />
          </svg>
        </a>
        <a
          aria-label="vktHomepage.ariaLabel"
          className="columns gapped-xxs"
          href="vktHomepage.link"
          rel="noreferrer"
          target="_blank"
        >
          links.vktHomepage.text
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="OpenInNewIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
            />
          </svg>
        </a>
        <div
          className="footer__container__links__contact"
        >
          <h2
            className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            links.contact.title
            :
          </h2>
          <a
            className="footer__container__links__contact__email"
            href="mailto:contactEmail"
            rel="noreferrer"
            target="_blank"
          >
            contactEmail
          </a>
        </div>
      </div>
      <div
        className="footer__container footer__container__contact-details"
      >
        <h2
          className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          address.name
        </h2>
        <br />
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          address.street
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          address.zipCity
        </p>
        <br />
        <div
          className="columns gapped-xxs"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 inline-text css-ahj2mt-MuiTypography-root"
          >
            address.phone.title
          </p>
          <a
            className="inline-text"
            href="tel:address.phone.number"
            rel="noreferrer"
            target="_blank"
          >
            address.phone.number
          </a>
        </div>
      </div>
    </div>
    <div
      className="footer__logo-row"
    >
      <div
        className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren footer__logo-row__divider css-1h4w97w-MuiDivider-root"
        role="separator"
      >
        <span
          className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
        >
          <div
            className="oph-logo columns"
          >
            <img
              alt="ophLogoAlt"
              className="footer__container__logo__oph"
              src="test"
            />
          </div>
        </span>
      </div>
    </div>
  </div>
</footer>
`;
